### PR TITLE
New script: linkml-run-examples. See #501

### PR DESCRIPTION
**Pull request by cmungall**
This adds a utility that can be used in any project
to ensure that all examples are valid and all counter
examples are invalid.

See:

 - #501

### CODE REVIEW

**Review by turbomam — COMMENTED** [review]
great. will try tomorrow.

File: linkml/workspaces/example_runner.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    schemaview = SchemaView(schema)
+    prefix_map = yaml.safe_load(open(prefixes)) if prefixes else None
+    runner = ExampleRunner(schemaview=schemaview,
+                           prefix_map=prefix_map,
+                           **kwargs)
+    runner.process_examples()
+
+
+if __name__ == "__main__":
+    cli(sys.argv[1:])
```
Comment (sujaypatil96) [inline linkml/workspaces/example_runner.py:206]: I don't think `sys` has been imported?

**Review by sujaypatil96 — APPROVED** [review]
Brilliant PR @cmungall. I think it's ready to go.🚀

File: linkml/workspaces/example_runner.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        input_dir = self.input_directory
+        counter_example_dir = self.counter_example_input_directory
+        if input_dir is None:
+            input_dir = Path.cwd() / "examples"
+        if counter_example_dir is None:
+            counter_example_dir = Path.cwd() / "counter_examples"
+        input_examples = glob.glob(os.path.join(str(input_dir), '*.yaml'))
+        input_counter_examples = glob.glob(os.path.join(str(counter_example_dir), '*.yaml'))
+        if not input_counter_examples:
+            logging.warning(f"No counter examples found in {self.counter_example_input_directory}")
```
[the remark this answers is not in the document]
Comment (cmungall) [reply linkml/workspaces/example_runner.py:96]: yes!

File: linkml/workspaces/example_runner.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    @property
+    def validator(self) -> DataValidator:
+        """
+        Get the current validator
+
+        :return:
+        """
+        if self._validator is None:
+            self._validator = JsonSchemaDataValidator(self.schemaview.schema)
+        return self._validator
```
[the remark this answers is not in the document]
Comment (cmungall) [reply linkml/workspaces/example_runner.py:73]: no this is only for gets - this should be specified at instantiation time

File: linkml/workspaces/example_runner.py (hunk 1 above)
Comment (cmungall) [reply linkml/workspaces/example_runner.py:206]: ah! let's do in a separate PR

**Pull request merged**